[DOCS] Build system hooks (#1230)
spatialos/gdk-for-unity: docs/modules/build-system/build-hooks.md

# Build hooks

> During a build, you can use the Unity provided callbacks to add pre- and post-processing to your worker build.

Reviewer: ```suggestion
During a build, you can use the Unity provided callbacks to add pre-processing and/or post-processing to your worker build.
```

> The build system feature module provides `WorkerBuilder.CurrentContext` containing information such as worker type and build options.

Reviewer: ```suggestion
The Build System Feature Module provides a static readonly variable, `WorkerBuilder.CurrentContext`, which contains information such as the worker type and build options.
```

I think our feature modules are proper nouns

> This context is only valid during a build.

Reviewer: ```suggestion
This context is only valid during a build. It will be `null` otherwise.
```

Author: It will throw otherwise

```csharp

> public class BuildPreProcessor : IPreprocessBuildWithReport, IPostprocessBuildWithReport

Reviewer: ```suggestion
public class BuildProcessor : IPreprocessBuildWithReport, IPostprocessBuildWithReport
```

It does both pre and post 😛

{
    public int callbackOrder => 0;
    public void OnPreprocessBuild(BuildReport report)
    {
        var workerType = WorkerBuilder.CurrentContext.WorkerType;
        Debug.Log($"Pre-processing build with worker type {workerType}");
    }

    public void OnPostprocessBuild(BuildReport report)
    {
        var workerType = WorkerBuilder.CurrentContext.WorkerType;
        Debug.Log($"Post-processing build with worker type {workerType}");
    }
}
```

> > Ensure that your Editor assembly definition references `Improbable.Gdk.BuildSystem`.

Reviewer: ```suggestion
> In order to access `WorkerBuilder.CurrentContext`, you must ensure that your Editor assembly definition references `Improbable.Gdk.BuildSystem`.
```
